run.go

```diff
@@ -588,60 +627,131 @@ loop:
 	}
 	fmt.Fprintf(color.Output, "\n")
 
+	// Print summary
+	summaryError := printSummary(fs, summaryFile, summaryFormat, engine, atTime)
+	if summaryError != nil {
+		log.WithError(summaryError).Errorf("Could not write summary to file %s", summaryFile)
+	}
+
+	if opts.Linger.Bool {
+		<-signals
+	}
+
+	if engine.IsTainted() {
+		return cli.NewExitError("", 99)
+	}
+	return nil
+}
+
+func printSummary(fs afero.Fs, filename string, summaryFormat SummaryFormat, engine *core.Engine, atTime time.Duration) error {
```

Comment:
I don't like that this does two things.

The cyclomatic complexity of actionRun is ridiculous, but it's something I'm going to do something about as soon as I can (we need proper packages!). Just have it call a `writeSummary()` function that writes to the file if one is given instead.